File: android/DateCalcualtor/src/jp/teres/numa/views/MainTabListener.scala
package jp.teres.numa.views

import android.content.Context
import android.app.ActionBar.{Tab , TabListener}
import android.app.{Fragment, FragmentTransaction}

class MainTabListener[T <: Fragment](ctx : Context,clazz : Class[T]) extends TabListener {

	private var mFragment : Fragment = _
	
	override def onTabReselected(tab : Tab, ft : FragmentTransaction) {}

	override def onTabSelected(tab : Tab, ft : FragmentTransaction) {
		mFragment = Option(mFragment).map {fr =>
			ft.attach(fr)
			fr
		}.getOrElse {
			val fragment = Fragment.instantiate(ctx, clazz.getName)
			ft.add(android.R.id.content, fragment)
			fragment
		}

	}

	override def onTabUnselected(tab : Tab, ft : FragmentTransaction) {
		Option(mFragment).foreach (ft.detach(_))
	}
} 

File: android/app/src/main/java/jp/teres/numa/DateCalculator/MainTabListener.java
package jp.teres.numa.DateCalculator;

import android.content.Context;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentTransaction;
import android.support.v7.app.ActionBar;

class MainTabListener<T extends Fragment> implements ActionBar.TabListener{

    final Context context;
    final Class<T> clazz;

    Fragment mFragment;

    public MainTabListener(Context context, Class<T> clazz) {
        this.context = context;
        this.clazz = clazz;
    }

    @Override
    public void onTabSelected(ActionBar.Tab tab, FragmentTransaction fragmentTransaction) {
        if (mFragment == null) {
            mFragment = Fragment.instantiate(context, clazz.getName());
            fragmentTransaction.add(android.R.id.content, mFragment);
        } else {
            fragmentTransaction.attach(mFragment);
        }
    }

    @Override
    public void onTabUnselected(ActionBar.Tab tab, FragmentTransaction fragmentTransaction) {
        if (mFragment == null) {
            return;
        }
        fragmentTransaction.detach(mFragment);
    }

    @Override
    public void onTabReselected(ActionBar.Tab tab, FragmentTransaction fragmentTransaction) {}
}
